pkg/apiserver/webhooks/datavolume-mutate.go

```diff
@@ -84,9 +85,14 @@ func (wh *dataVolumeMutatingWebhook) Admit(ar admissionv1.AdmissionReview) *admi
 		}
 		dataSource, err := wh.cdiClient.CdiV1beta1().DataSources(ns).Get(context.TODO(), dataVolume.Spec.SourceRef.Name, metav1.GetOptions{})
 		if err != nil {
-			return toAdmissionResponseError(err)
+			if !k8serrors.IsNotFound(err) {
+				return toAdmissionResponseError(err)
+			}
+			// Create a mock 'pvcSource' to do the validation even if the dataSource doesn't exist yet
+			pvcSource = &cdiv1.DataVolumeSourcePVC{Name: dataVolume.Spec.SourceRef.Name, Namespace: ns}
```

Comment:
There is no way to securely allow creation of a DataVolume with sourceRef that does not exist.  In the webhook, we must validate that the user creating the DataVolume has permission to the source PVC namespace.  But there is no way to know what the source namespace may be.  And this validation cannot be deferred to later (in the controller) because the user context is lost.

Reply:
You are right, I somehow assumed that the datasource's namespace would be the same as the source PVC's one. However, I'm still trying to figure out if there's a way to do this validation in the controller, or if we could just skip it and let the clone fail if the DataVolume doesn't meet the requirements.

Reply:
I've just seen that `clone.CanUserClonePVC` doesn't check if the source namespace exists, so I'll also modify that.